pkg/utils/utils.go

```diff
@@ -44,3 +45,16 @@ func ContextWithDeadlineFn(ctx context.Context, deadlineFn func(orig time.Time)
 	}
 	return ctx, cancel
 }
+
+func IsZero[C comparable](val C) bool {
+	var zero C
+	return zero == val
+}
+
+// AddHexPrefix adds the prefix (0x) to a given hex string.
+func AddHexPrefix(str string) string {
```

Comment:
How about `Ensure` since it doesn't add when it's already there.
```suggestion
// EnsureHexPrefix adds the prefix (0x) to a given hex string.
func EnsureHexPrefix(str string) string {
```